src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test/TelemetryReporterTests.cs

```diff
@@ -322,4 +324,98 @@ public void TrackLspRequest()
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var nestedException = CreateNestedException(exceptionType, depth);
+
+        // Act
+        reporter.ReportFault(nestedException, "Test message");
+
+        // Assert
+        AssertNestedFaultReport(exceptionType, depth, reporter);
+    }
+
+    [Theory]
+    [InlineData(typeof(OperationCanceledException), RecursionDepthLimit)]
+    [InlineData(typeof(AggregateException), RecursionDepthLimit)]
+    [InlineData(typeof(Exception), RecursionDepthLimit)]
+    // Prevent stack overflow exception crash when depth beyond the limit
+    [InlineData(typeof(OperationCanceledException), 40000)]
+    [InlineData(typeof(AggregateException), 40000)]
+    [InlineData(typeof(Exception), 40000)]
+    public void ReportFault_WithRemoteInvocationException_ContainsNestedException_MethodCompletesWithoutException(Type exceptionType, int depth)
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var nestedException = new RemoteInvocationException("Test", 0, CreateNestedException(exceptionType, depth));
+
+        // Act
+        reporter.ReportFault(nestedException, nestedException.Message);
+
+        // Assert
+        AssertNestedFaultReport(exceptionType, depth, reporter);
+    }
+
+    private static void AssertNestedFaultReport(Type exceptionType, int depth, TestTelemetryReporter reporter)
+    {
+        if (exceptionType == typeof(OperationCanceledException) && depth > RecursionDepthLimit)
+        {
+            Assert.Empty(reporter.Events);
+        }
+        else
+        {
+            Assert.Collection(reporter.Events,
+                e1 =>
+                {
+                    Assert.Equal("dotnet/razor/fault", e1.Name);
+                });
+        }
+    }
+
+    private Exception CreateNestedException(Type exceptionType, int depth)
+    {
+        var exception = CreateException(exceptionType, "Test", null);
+        for (var i = 0; i < depth; i++)
+        {
+            exception = CreateException(exceptionType, "Test", exception);
+        }
+
+        return exception;
+    }
+
+    private Exception CreateException(Type exceptionType, string message, Exception? innerException)
+    {
+        if (exceptionType == typeof(OperationCanceledException))
+        {
+            return new OperationCanceledException(message, innerException);
+        }
+        else if (exceptionType == typeof(AggregateException))
+        {
+            if (innerException == null)
+            {
+                return new AggregateException(message, new Exception());
+            }
+
+            return new AggregateException(message, innerException);
+        }
+        else if (exceptionType == typeof(InvalidOperationException))
+        {
+            return new InvalidOperationException(message, innerException);
+        }
```

Comment:
There doesn't seem to be any tests that use this exception type, nor any code in the telemetry reporter to extract inner exceptions from this exception type. Is that done somewhere else?

Reply:
I'll remove this